system/ee/legacy/database/DB_active_rec.php

```diff
@@ -302,15 +303,21 @@ public function join($table, $cond, $type = '')
         $this->_track_aliases($table);
 
         // Strip apart the condition and protect the identifiers
-        if (preg_match('/([\w\.]+)([\W\s]+)(.+)/', $cond, $match)) {
+        if (preg_match('/([\w\.]+)([\W\s]+)(.+)/', $cond, $match) && !$alias) {
```

Comment:
It might be better to check `$alias` first (and maybe check with is_empty - see the comment below) to save a little bit by avoiding preg_match processing if it's not needed